hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestFileSizeCountTask.java

```diff
@@ -73,45 +73,60 @@ public void setUp() {
   }
 
   @Test
-  public void testReprocess() throws IOException {
-    OmKeyInfo omKeyInfo1 = mock(OmKeyInfo.class);
-    given(omKeyInfo1.getKeyName()).willReturn("key1");
-    given(omKeyInfo1.getVolumeName()).willReturn("vol1");
-    given(omKeyInfo1.getBucketName()).willReturn("bucket1");
-    given(omKeyInfo1.getDataSize()).willReturn(1000L);
-
-    OmKeyInfo omKeyInfo2 = mock(OmKeyInfo.class);
-    given(omKeyInfo2.getKeyName()).willReturn("key2");
-    given(omKeyInfo2.getVolumeName()).willReturn("vol1");
-    given(omKeyInfo2.getBucketName()).willReturn("bucket1");
-    given(omKeyInfo2.getDataSize()).willReturn(100000L);
-
-    OmKeyInfo omKeyInfo3 = mock(OmKeyInfo.class);
-    given(omKeyInfo3.getKeyName()).willReturn("key3");
-    given(omKeyInfo3.getVolumeName()).willReturn("vol1");
-    given(omKeyInfo3.getBucketName()).willReturn("bucket1");
-    given(omKeyInfo3.getDataSize()).willReturn(1125899906842624L * 4); // 4PB
+  public void testReprocessFSO1() throws IOException {
+    OmKeyInfo[] omKeyInfos = new OmKeyInfo[3];
+    String[] keyNames = {"key1", "key2", "key3"};
+    String[] volumeNames = {"vol1", "vol1", "vol1"};
+    String[] bucketNames = {"bucket1", "bucket1", "bucket1"};
+    Long[] dataSizes = {1000L, 100000L, 1125899906842624L * 4};
+
+    // Loop to initialize each instance of OmKeyInfo
+    for (int i = 0; i < 3; i++) {
+      omKeyInfos[i] = mock(OmKeyInfo.class);
+      given(omKeyInfos[i].getKeyName()).willReturn(keyNames[i]);
+      given(omKeyInfos[i].getVolumeName()).willReturn(volumeNames[i]);
+      given(omKeyInfos[i].getBucketName()).willReturn(bucketNames[i]);
+      given(omKeyInfos[i].getDataSize()).willReturn(dataSizes[i]);
+    }
 
+    // Create two mock instances of TypedTable, one for FILE_SYSTEM_OPTIMIZED
+    // layout and one for LEGACY layout
     OMMetadataManager omMetadataManager = mock(OmMetadataManagerImpl.class);
-    TypedTable<String, OmKeyInfo> keyTable = mock(TypedTable.class);
-
-    TypedTable.TypedTableIterator mockKeyIter = mock(TypedTable
-        .TypedTableIterator.class);
-    TypedTable.TypedKeyValue mockKeyValue = mock(
-        TypedTable.TypedKeyValue.class);
-
-    when(keyTable.iterator()).thenReturn(mockKeyIter);
-    when(omMetadataManager.getKeyTable(getBucketLayout())).thenReturn(keyTable);
-    when(mockKeyIter.hasNext())
-        .thenReturn(true)
-        .thenReturn(true)
-        .thenReturn(true)
-        .thenReturn(false);
-    when(mockKeyIter.next()).thenReturn(mockKeyValue);
-    when(mockKeyValue.getValue())
-        .thenReturn(omKeyInfo1)
-        .thenReturn(omKeyInfo2)
-        .thenReturn(omKeyInfo3);
+    TypedTable<String, OmKeyInfo> keyTable1 = mock(TypedTable.class);
+    TypedTable<String, OmKeyInfo> keyTable2 = mock(TypedTable.class);
```

Comment:
names

Reply:
Thanks for the suggestion, i'll make the necessary changes!

Reply:
@GeorgeJahad can you please take a look!